src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -298,20 +304,24 @@ public override object Clone()
             //  filterrate = pto->minFeeFilter;
             //}
 
-            List<TxMempoolInfo> sends = await this.manager.MempoolLock.WriteAsync(() =>
+            var sends = new List<TxMempoolInfo>();
+            lock (this.lockObject)
```

Comment:
Perhaps this could be done with minimal time inside the lock?
You can do the logic in the foreach outside of the lock (say with local lists) and the also do the tracing, then inside the lock just copy/remove from the shared resource.